CDCgov/prime-reportstream
# #2455: Updates to consolidated error and warning messaging

This PR updates the consolidated error/warning messaging according to Issue #2455 

Test Steps:
1. pull `kevinhaube/2455-fix-error-messages`
2. `./gradlew clean package` && `./devenv-infrastructure.sh restart`
3. Use `curl` to post a CSV with manipulated fields to trigger errors/warnings
4. Observe the response and confirm errors/warnings were reported accurately

Note: `itemDetails` and `itemNums` should only show for `ITEM` scope error/warning
Note: `itemDetails` should only show when `verbose`

## Changes
- Added `scope` to the consolidated errors/warnings
- `rows` renamed to `itemNums`
- `itemDetails` available when verbose
- Removed old error/warning handling and renamed consolidated arrays to `errors` and `warnings`

## Checklist

### Testing

### Process

## Fixes
- #2455 

## Specific Security-related subjects a reviewer should pay specific attention to
- Does this PR introduce new endpoints?
No
- Does this PR include changes in authentication and/or authorization of existing endpoints?
No
- Does this change introduce new dependencies that need vetting?
No
- Does this change require changes to our infrastructure?
No
- Does logging contain sensitive data?
No
- Does this PR include or remove any sensitive information itself?
No

If you answered '_yes_' to any of the questions above, conduct a detailed Review that addresses at least:

- What are the potential security threats and mitigations? Please list the _STRIDE_ threats and how they are mitigated
    - **S**poofing (faking authenticity)
        - Threat _T_, which could be achieved by _A_, is mitigated by _M_
    - **T**ampering (influence or sabotage the integrity of information, data, or system)
    - **R**epudiation (the ability to dispute the origin or originator of an action)
    - **I**nformation disclosure (data made available to entities who should not have it)
    - **D**enial of service (make a resource unavailable)
    - **E**levation of Privilege (reduce restrictions that apply or gain privileges one should not have)
- Have you ensured logging does not contain sensitive data?
- Have you received any additional approvals needed for this change?
